stores/autopilot.go

```diff
@@ -2,37 +2,30 @@ package stores
 
 import (
 	"context"
-	"errors"
 
 	"go.sia.tech/renterd/api"
 	sql "go.sia.tech/renterd/stores/sql"
 )
 
-func (s *SQLStore) Autopilots(ctx context.Context) (aps []api.Autopilot, _ error) {
+func (s *SQLStore) AutopilotState(ctx context.Context) (ap api.AutopilotState, _ error) {
 	err := s.db.Transaction(ctx, func(tx sql.DatabaseTx) (err error) {
-		aps, err = tx.Autopilots(ctx)
-		return
-	})
-	return aps, err
-}
-
-func (s *SQLStore) Autopilot(ctx context.Context, id string) (ap api.Autopilot, _ error) {
-	err := s.db.Transaction(ctx, func(tx sql.DatabaseTx) (err error) {
-		ap, err = tx.Autopilot(ctx, id)
+		ap, err = tx.AutopilotState(ctx)
 		return
 	})
 	return ap, err
 }
 
-func (s *SQLStore) UpdateAutopilot(ctx context.Context, ap api.Autopilot) error {
-	// validate autopilot
-	if ap.ID == "" {
-		return errors.New("autopilot ID cannot be empty")
-	}
-	if err := ap.Config.Validate(); err != nil {
+func (s *SQLStore) UpdateAutopilotConfig(ctx context.Context, cfg api.AutopilotConfig) error {
+	if err := cfg.Hosts.Validate(); err != nil {
```

Comment:
Kind of surprising we didn't do that but we should also validate the contracts config at some point.